From 4e17f500b3c5a95b6a42fc9e8b0ab0287ab5201a Mon Sep 17 00:00:00 2001
From: Maharaja Kennadyrajan <quic_mkenna@quicinc.com>
Date: Mon, 9 Dec 2024 16:40:11 +0530
Subject: [PATCH] UPSTREAM:PROTOCOL: wpa_supplicant: Add mesh EHT320 support
 and capabilities

Add the mesh EHT320 support and EHT capabilites which are removed
from the patch q02-045-mesh_add_EHT_support.patch.

Signed-off-by: Maharaja Kennadyrajan <quic_mkenna@quicinc.com>
---
 wpa_supplicant/ap.c       | 4 ++++
 wpa_supplicant/mesh.c     | 3 +++
 wpa_supplicant/mesh_mpm.c | 5 +++++
 3 files changed, 12 insertions(+)

diff --git a/wpa_supplicant/ap.c b/wpa_supplicant/ap.c
index 7d07017..57567e8 100644
--- a/wpa_supplicant/ap.c
+++ b/wpa_supplicant/ap.c
@@ -446,6 +446,10 @@ int wpa_supplicant_conf_ap_ht(struct wpa_supplicant *wpa_s,
 			    ssid->he)
 				conf->ieee80211ax = 1;
 
+			if (mode->eht_capab[wpas_mode_to_ieee80211_mode(
+			    ssid->mode)].eht_supported && ssid->eht)
+				conf->ieee80211be = 1;
+
 			if (mode->vht_capab && ssid->vht) {
 				conf->ieee80211ac = 1;
 				conf->vht_capab |= mode->vht_capab;
diff --git a/wpa_supplicant/mesh.c b/wpa_supplicant/mesh.c
index 093e0fc..9120a14 100644
--- a/wpa_supplicant/mesh.c
+++ b/wpa_supplicant/mesh.c
@@ -471,6 +471,9 @@ static int wpa_supplicant_mesh_init(struct wpa_supplicant *wpa_s,
 		case 160:
 			conf->op_class = 134;
 			break;
+		case 320:
+			conf->op_class = 137;
+			break;
 		default:
 			conf->op_class = 131;
 			break;
diff --git a/wpa_supplicant/mesh_mpm.c b/wpa_supplicant/mesh_mpm.c
index aacf751..3250a85 100644
--- a/wpa_supplicant/mesh_mpm.c
+++ b/wpa_supplicant/mesh_mpm.c
@@ -798,6 +798,11 @@ static struct sta_info * mesh_mpm_add_peer(struct wpa_supplicant *wpa_s,
 			   elems->eht_capabilities_len);
 #endif /*CONFIG_IEEE80211BE */
 
+#ifdef CONFIG_IEEE80211BE
+	copy_sta_eht_capab(data, sta, IEEE80211_MODE_MESH, elems->he_capabilities,
+			   elems->he_capabilities_len, elems->eht_capabilities, elems->eht_capabilities_len);
+#endif /* CONFIG_IEEE80211BE */
+
 	if (hostapd_get_aid(data, sta) < 0) {
 		wpa_msg(wpa_s, MSG_ERROR, "No AIDs available");
 		ap_free_sta(data, sta);
-- 
2.34.1

